sqlsorcery.py

```diff
@@ -0,0 +1,100 @@
+from os import getenv
+from google.oauth2 import service_account
+import pandas as pd
+import pandas_gbq
+from sqlalchemy import create_engine, Table, MetaData
+from sqlalchemy.engine.url import URL
+from sqlalchemy.exc import NoSuchTableError
+from sqlalchemy.schema import DropTable
+
+
+class Connection:
+    def __init__(
+        self,
+        dialect=None,
+        user=None,
+        pwd=None,
+        server=None,
+        port=None,
+        db=None,
+        schema=None,
+        creds=None,
+    ):
+        self.dialect = dialect
+        self.schema = schema
+        self.db = db
+        self.options = {
+            "username": user,
+            "password": pwd,
+            "host": server,
+            "port": port,
+            "database": db,
+        }
+        self.creds = creds
+        self.helpers = {}
+        self.engine = self._get_engine()
+
+    def _cstr(self):
+        return URL(self.dialect, **self.options)
+
+    def _bigquery_options(self):
+        if self.dialect == "bigquery":
+            self._get_creds()
+            self.helpers["credentials_path"] = self.creds
+
+    def _get_creds(self):
+        credentials = service_account.Credentials.from_service_account_file(self.creds)
+        pandas_gbq.context.credentials = credentials
+        pandas_gbq.context.project = self.db
+
+    def _get_engine(self):
+        self._bigquery_options()
+        return create_engine(self._cstr(), **self.helpers)
+
+    def query(self, sql_query, params=None):
+        if self.dialect == "bigquery":
```

Comment:
With all of the if statements, we should probably do one of:
 - store bigquery and the other options as constants somewhere to reference. If we foresee any kind of other customizations by dialect we may want to go the enum route eventually.
 - move this to a function so it is only written once.

Reply:
@zkagin Could you provide a code snippet or example of your thinking here? I would like to reduce the need to keep checking for the dialect (here and in the sqlsorcery library updates I'm making), but I can't wrap my head around how this would be done without it being on a function by function basis.

Reply:
I think the way to eventually handle this is probably something like the code below, but for now (and in this diff) it isn't necessary.
```
constants.py
from enum import Enum
class Dialect(Enum):
    BigQuery
    Postgres
    MSSql
```

Another shortcut would be to have an `_is_bigquery()` function so that you're not typing out an exact string each time, though that would be a temporary measure.

I'm going to look into doing the first solution in the other repo — I think this is fine to leave as is right now.